jvb-api/jvb-api-client/src/main/kotlin/org/jitsi/videobridge/api/client/v1/JvbApi.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@file:Suppress("unused")
+
+package org.jitsi.videobridge.api.client.v1
+
+import io.ktor.client.HttpClient
+import io.ktor.client.engine.cio.CIO
+import io.ktor.client.features.websocket.WebSockets
+import org.jitsi.utils.logging2.LoggerImpl
+import org.jitsi.videobridge.api.util.SmackXmlSerDes
+import org.jitsi.videobridge.api.util.SynchronousWebSocketClient
+import org.jivesoftware.smack.packet.IQ
+
+/**
+ * JVB Client API for controlling a JVB instance
+ */
+class JvbApi(jvbHost: String, jvbPort: Int) {
+    private val client = HttpClient(CIO) {
+        install(WebSockets)
+    }
+
+    private val wsClient = SynchronousWebSocketClient(
+        client,
+        host = jvbHost,
+        port = jvbPort,
+        path = "/v1/ws",
+        parentLogger = LoggerImpl(JvbApi::class.qualifiedName)
+    ).also {
+        it.run()
+    }
+
+    /**
+     * Send an [IQ] and return the response [IQ].  Call is synchronous.
+     */
+    fun sendIqAndGetReply(iq: IQ): IQ {
+        return wsClient.sendIqAndGetReply(iq)
+    }
+
+    /**
+     * Send an [IQ] and don't wait for its response
+     */
+    fun sendAndForget(iq: IQ) {
+        wsClient.sendIqAndForget(iq)
+    }
+}
+
+/**
+ * Send [IQ]s and get their replies via a [SynchronousWebSocketClient].
+ */
+private fun SynchronousWebSocketClient.sendIqAndGetReply(iq: IQ): IQ {
+    val resp = sendAndGetReply(SmackXmlSerDes.serialize(iq))
```

Comment:
This doesn't timeout, as, from what I can tell, the XMPP version didn't.  Do you mean if the connection closes?  That would be a good thing to add a test for, I'll look at that.

Reply:
So it's a bit awkward to add a unit tests for this since there's a blocking call involved (there will be races between the blocking thread and the one calling cancel).  But, I did write one for testing and cleaned this up a bit by throwing an exception if things go south while waiting for the response.

Reply:
The XMPP implementation does timeout and return `null` when it does:
https://github.com/jitsi/jicofo/blob/master/src/main/java/org/jitsi/impl/protocol/xmpp/XmppProtocolProvider.java#L644

Reply:
Thanks for the pointer, I'll add a timeout.